### PR TITLE
[Misc] Disable LibXSMM SpMM Cmp for AVX2 platforms

### PR DESCRIPTION
**Pull request by itaraban**
## Description

Unfortunately SpMM Cmp was disabled for AVX2 and less platforms in [LIBXSMM](https://github.com/libxsmm/libxsmm/blob/main_stable/src/generator_mateltwise_sse_avx_avx512.c#L746)
Resolves https://github.com/dmlc/dgl/issues/5960
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: src/array/cpu/spmm.h [hunk 1]
```diff
@@ -267,10 +267,10 @@ void SpMMCmpCsr(
 #if !defined(_WIN32)
 #ifdef USE_LIBXSMM
   int cpu_id = libxsmm_cpuid_x86();
-  const bool no_libxsmm =
-      bcast.use_bcast || std::is_same<DType, double>::value ||
-      (std::is_same<DType, BFloat16>::value && cpu_id < LIBXSMM_X86_AVX512) ||
```
Comment (chang-l) [inline src/array/cpu/spmm.h:272]: I see you removed BF16 check `std::is_same<DType, BFloat16>::value`. Is it an overlap with / equivalent to `cpu_id < LIBXSMM_X86_AVX512)`? Would it reopen this issue https://github.com/dmlc/dgl/issues/5647 on some types of CPU?

Comment (itaraban) [reply src/array/cpu/spmm.h:272]: Previously this path was broken for bf16 data type and avx2, now it is broken for all data types.
That's why I modified this line, to disable `SpMMCmpCsrLibxsmm` for platforms without AVX512

**Comment by itaraban** [comment]
@dgl-bot

File: src/array/cpu/spmm.h (hunk 1 above)
Comment (chang-l) [reply src/array/cpu/spmm.h:272]: I see. To confirm, SpMMCmpCsrLibxsmm works with BF16 with AVX512 on all platforms?

Comment (itaraban) [reply src/array/cpu/spmm.h:272]: `SpMMCmpCsrLibxsmm` for all data types works only for AVX512 platforms

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**